compiler warning (macOS) in `src/cgmemmgr.cpp`:

Would be nice to get rid of this and get back to a clean build process:

```c++
    CC src/cgmemmgr.o
/Users/stefan/projects/julia/src/cgmemmgr.cpp:742:10: warning: '(anonymous
      namespace)::RTDyldMemoryManagerJL::notifyObjectLoaded' hides overloaded virtual
      function [-Woverloaded-virtual]
    void notifyObjectLoaded(RuntimeDyld &Dyld,
         ^
/Users/stefan/projects/julia/usr/include/llvm/ExecutionEngine/RTDyldMemoryManager.h:46:16: note:
      hidden overloaded virtual function 'llvm::MCJITMemoryManager::notifyObjectLoaded'
      declared here: type mismatch at 1st parameter ('llvm::ExecutionEngine *' vs
      'llvm::RuntimeDyld &')
  virtual void notifyObjectLoaded(ExecutionEngine *EE,
               ^
1 warning generated.
```

cc: @yuyichao

--- a/src/cgmemmgr.cpp
+++ b/src/cgmemmgr.cpp
@@ -739,6 +739,7 @@ public:
                                  unsigned SectionID, StringRef SectionName,
                                  bool isReadOnly) override;
 #if JL_LLVM_VERSION >= 30800
+    using SectionMemoryManager::notifyObjectLoaded;
     void notifyObjectLoaded(RuntimeDyld &Dyld,
                             const object::ObjectFile &Obj) override;
 #endif

